### ISSUE TITLE
Auto Workbench will accept items not used in its recipe via automation

### ISSUE THREAD

**Issue by LuigiHutch** [BuildCraft/BuildCraft#3989]
BuildCraft version: 7.99.15-pre4
Forge version: 1.12.2-14.23.1.2590
Link to crash report or log: N/A
Singleplayer or multiplayer: Both

Items that are not used in the auto workbench recipe are able to enter via automation. I tested with hoppers and pipes where neither were able to put items that are not in the recipe into the auto-workbench. This is one of the setups I used:
![2018-01-21_11 38 14](https://user-images.githubusercontent.com/9326429/35193777-ae9e810e-fe9f-11e7-8cc4-230e0284469d.png)
![2018-01-21_11 38 16](https://user-images.githubusercontent.com/9326429/35193778-aeb9e8ae-fe9f-11e7-9b29-e9c57506c3fd.png)

Basically https://github.com/BuildCraft/BuildCraft/issues/3334 in 7.99.15-pre4

**Comment by SandGrainOne** [comment]
This is one of those things I think the player could have solved themselves instead of having BuildCraft solve it for them.
